teachertool/src/components/PrintButton.tsx

```diff
@@ -1,17 +1,23 @@
 import * as React from "react";
-import { useRef } from "react";
 import { useReactToPrint } from "react-to-print";
 import { Toolbar } from "./Toolbar";
+import { stateAndDispatch } from "../state";
+import { showToast } from "../state/actions";
+import { makeToast } from "../utils";
+import { Strings } from "../constants";
 
 interface PrintButtonProps {
     printRef: React.RefObject<HTMLDivElement>;
-    onHandlePrint: () => void;
 }
 
-export const PrintButton: React.FC<PrintButtonProps> = ({ printRef, onHandlePrint }) => {
+export const PrintButton: React.FC<PrintButtonProps> = ({ printRef }) => {
+    const { state: teacherTool } = stateAndDispatch();
     const handlePrint = useReactToPrint({
         content: () => printRef.current,
-        onAfterPrint: onHandlePrint,
+        onPrintError: () => showToast(makeToast("error", lf("Unable to print evaluation results"), 2000)),
+        documentTitle: `MakeCode Project Evaluation Results - ${pxt.Util.sanitizeFileName(
```

Comment:
Including the rubric name in the document title would be helpful, I think.

```suggestion
        documentTitle: `MakeCode - ${rubric name} - 
```

Reply:
Yeah, I was torn about this. I think there are two scenarios to consider with the naming of the file:
1. What will make the teacher's job easier to find a specific student's submission results in a whole slew of files?
2. If students receive the file online, with the associated file name, do they need to know the rubric name?

Is the suggestion to include both the rubric and file name? So something like `MakeCode - My Rubric - Rietkerk Rock Paper Scissors`? I do think having something like that would help teachers, but it also seems really long. To be fair, what I had before was long, too. And if they gave the electronic file back to students, I would be under the impression that teachers would take out the rubric name, but maybe that wouldn't happen. I think I'd prefer to get user feedback on this and change it to something more permanent then, but maybe have the rubric and project name in for the time being?

Reply:
I think you could get rid of the MakeCode and just have "Rubric - Project", personally. But I also don't think it's a big deal either way.

Reply:
So if I go this route, I shouldn't localize anything, right?

Reply:
"Rubric - Project" seems like a good doc name. No localization needed either.